src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/PlayerDeathSystem.java

```diff
@@ -90,6 +90,9 @@ private void dropItemsFromInventory(EntityRef player) {
 
     private void updateStatistics(EntityRef player, String type) {
         PlayerStatisticsComponent playerStatisticsComponent = player.getComponent(PlayerStatisticsComponent.class);
+        if (player.getId() == 0) {
```

Comment:
When is the player ID 0 and why do we want to simply return in that case?
And what happens if a player without ID 0 dies on their own account but doesn't have a `PlayerStatisticsComponent` yet?

Reply:
let me give a brief explanation of what I have done. So, when the player dies by itself, `beforeDestroy()` calls `updateStatistics()`. 

The first time `updateStatistics()` is called with parameter `event.instigator` (which will be NullRef in this case as player dies by itself, returning with EntityRef of ID=0) and `"kills"`. So, in this case we cannot update statistics.

The Second time when `updateStatistics()` is called with parameter `player` and `"deaths"`, we get `EntityRef` of the player (initially having `PlayerStatisticsComponent`). So, in this case we update the player's deaths.

Answer 1: Player ID=0 when `event.instigator` is called and player dies by itself. We return in that case because we cannot update the statistics of NullRef (also no point of updating any stats here)

Answer 2: The player gets `PlayerStatisticsComponent` as soon as the player gets spawned on the fool's platform

Reply:
ok, will investigate on this issue  👍